DEVELOPMENT.md

```diff
@@ -0,0 +1,74 @@
+# Setting up newsroom for development
+
+## Repositories
+
+Use `develop` branch of `newsroom-core` repository and `master` branch of `superdesk-app`.
+
+## Install dependencies
+
+* docker engine - follow instructions at https://docs.docker.com/engine/install/ubuntu
+* `sudo apt install docker-compose`
+* `sudo apt install git`
+* `sudo apt install python3.10-venv` (may vary depending on python version)
+* `sudo apt-get install python3-dev`
+* `sudo apt install nodejs`
+* `sudo apt install npm`
+* `sudo npm install -g n`
+* use node 14 - `sudo n 14`
+
+## Setup and link newsroom-core client
+* navigate to `newsroom-core`
+* `npm install`
+* `npm link`
+
+## Start client (needs to be started before back-end server)
+* navigate to `newsroom-app/client`
+* `npm install`
+* `npm link newsroom-core`
+* `npm start`
+
+## Starting required services via docker (redis, mongo, elastic)
+* navigate to `newsroom-app`
+* `cd server`
+* `sudo docker-compose up redis mongo elastic`
+
+## Create .env file
+* navigate to `newsroom-app/server`
+* create .env file with the following contents:
+
+```
+WEBPACK_ASSETS_URL=http://localhost:8080
+WEBPACK_SERVER_URL=http://localhost:8080
+SECRET_KEY=newsroom
+```
+
+## Starting local server
+* navigate to `newsroom-app`
+* make sure you're on `master` branch
+* `cd server`
+* create virtual environment `python3 -m venv env`
+* activate virtual environment `source env/bin/activate`
+* install python dependencies - `pip install -r dev-requirements.txt`
+* link local server - `pip install -Ue <path-to-newsroom-core>`
+* create a user - `python manage.py create_user admin@localhost.com admin John Doe true`
```

Comment:
We could base this on what we use in FireQ instances (see [prepopulate](https://github.com/superdesk/fireq/blob/master/tpl/newsroom-app/prepopulate.sh) script).
```bash
python manage.py initialize_data
python manage.py data_upgrade
python manage.py create_user admin@example.com admin Admin Admin true || :
python manage.py schema_migrate || true
```

- This creates the user`admin@example.com` instead of `admin@localhost.com` (same user email as our test instances)
- Running `initialize_data` prepares the system (including elastic)
- Running `data_upgrade` runs any migration scripts
- Running `schema_migrate` runs Mongo/Elastic schema migrations

These last 2 above are important to run when there are changes to the DB schema (see [SCHEMA_VERSIONS](https://github.com/superdesk/newsroom-core/blob/develop/newsroom/__init__.py#L19))